vault/token_store.go

```diff
@@ -2441,6 +2442,24 @@ func (ts *TokenStore) handleCreateCommon(ctx context.Context, req *logical.Reque
 		}
 	}
 
+	// It is allowed to overwrite the corresponding entity of a token.
+	// This is only possible when client token has sudo/root.
+	if data.EntityID != "" && ts.core.identityStore != nil {
+		if !isSudo {
+			return logical.ErrorResponse("root or sudo privileges required to specify entity id"), logical.ErrInvalidRequest
+		}
+		entity, err := ts.core.identityStore.MemDBEntityByID(data.EntityID, false)
+		if err != nil {
+			return logical.ErrorResponse(err.Error()), logical.ErrInvalidRequest
```

Comment:
The second return parameter can be `nil` here as its not an invalid request. It's an internal error.